module/tasks/common.tasks.ps1

```diff
@@ -18,7 +18,33 @@ task EnsureGitHubCli -If { !$SkipEnsureGitHubCli } -After InitCore {
 }
 
 # Synopsis: Installs and imports the specified PowerShell modules. ***NOTE**: PowerShell repositories will be trusted by default.*
-task setupModules -If { $null -ne $RequiredPowerShellModules -and $RequiredPowerShellModules -ne @{} } {
+task setupModules -If { $null -ne $RequiredPowerShellModules -and $RequiredPowerShellModules -ne @{} } -After InitCore {
     Install-PSResource -RequiredResource $RequiredPowerShellModules -Scope CurrentUser -TrustRepository | Out-Null
     $RequiredPowerShellModules.Keys | ForEach-Object { Import-Module $_ }
 }
+
+
+# Support the special InvokeBuild entry & exit actions in a way that allows
+# extensions to register their own actions and have them run using this
+# built-in InvokeBuild functionality.  This is primarily intended to support
+# compensation scenarios in the event of terminating errors.
+Enter-Build {
```

Comment:
What's the failure mode of these extensibility points? If there's an error do you want everything to stop, or do you want the script to continue (i.e. `-ErrorAction Continue`)?

Reply:
For the `Enter-Build` scenario my thinking was that we would just adhere to whatever the calling script had defined (i.e. generally `-ErrorAction Stop`).  I made it explicit for the `Exit-Build` because it's important that any compensation tasks have the opportunity to run.

Reply:
I've fixed the confusing comment!